soc/arm/microchip_mec/common/spigen/mec_spi_gen.py

```diff
@@ -0,0 +1,476 @@
+                        dest="spi_slew_rate",
+                        choices = ["slow", "fast"],
+                        default="slow",
+                        help="SPI pins slew rate")
+    parser.add_argument("--fill",
+                        dest="fill",
+                        action='store_true',
+                        help="Fill with 0xFF to flash size")
+    parser.add_argument("-v",
+                        dest="verbose",
+                        action='store_true',
+                        help="Enable messages to console")
+
+    ret_args = parser.parse_args()
+
+    return ret_args
+
+def main():
+    """MEC SPI Gen"""
+    args = parse_args()
+
+    verbose_mode = args.verbose
+
+    if verbose_mode:
+        print("Command line arguments/defaults")
+        print("  chip = {0}".format(args.chip))
+        print("  infilename = {0}".format(args.infilename))
+        print("  outfilename = {0}".format(args.outfilename))
+        print("  SPI size (kilobytes) = {0}".format(args.spi_size_kb))
+        print("  Entry point address = {0}".format(args.entry_point))
+        print("  SPI frequency MHz = {0}".format(args.spi_freq_mhz))
+        print("  SPI Read Mode = {0}".format(args.spi_read_mode))
+        print("  SPI Signalling Mode = {0}".format(args.spi_mode))
+        print("  SPI drive strength = {0}".format(args.spi_drive_strength))
+        print("  SPI slew rate fast = {0}".format(args.spi_slew_rate))
+        print("  Verbose = {0}".format(args.verbose))
+
+    if args.infilename is None:
+        if verbose_mode:
+            print("ERROR: Specify input binary file name with -i")
+        sys.exit(-1)
+
+    if args.outfilename is None:
+        if verbose_mode:
+            print("ERROR: Specify output binary file name with -o")
+        sys.exit(-1)
+
+    chip = args.chip
+    spi_read_mode = args.spi_read_mode
+    spi_freq_mhz = args.spi_freq_mhz
+    spi_mode = args.spi_mode
+    spi_drive_str_mult = args.spi_drive_strength
+    spi_slew = args.spi_slew_rate
+
+    spi_size = args.spi_size_kb * 1024
+
+    indata = None
+    with open(args.infilename, "rb") as fin:
+        indata = fin.read()
+
+    indata_len = len(indata)
+    if verbose_mode:
+        print("Read input FW binary: length = {0}".format(indata_len))
+
+    # if necessary pad input data to PLD_GRANULARITY required by Boot-ROM loader
+    if (indata_len % PLD_GRANULARITY) != 0:
+        pad_len = PLD_GRANULARITY - (indata_len % PLD_GRANULARITY)
+        # NOTE: MCHP Production SPI Image Gen. pads with 0
+        padding = PLD_PAD_BYTE * pad_len
+        indata = indata + padding
+
+    if verbose_mode:
+        print("Padded FW binary: length = {0}".format(len(indata)))
+
+    indata_len = len(indata)
+
+    # Do we have enough space for 4KB block containing TAG and Header, padded FW binary,
+    # EC Info Block, Co-Sig Block, and Trailer?
+    if indata_len > (spi_size - (PLD_SPI_LOC + EC_INFO_BLOCK_SIZE + COSIG_SIZE + TRAILER_SIZE)):
```

Comment:
`spi_size - (PLD_SPI_LOC + EC_INFO_BLOCK_SIZE + COSIG_SIZE + TRAILER_SIZE` could probably use an intermediate varaible for clarity, something like available_size? So that it can be understood without the comment

Reply:
added intermediate variable